src/Fantomas/CodePrinter.fs

```diff
@@ -5418,14 +5425,17 @@ and genSynBindingValue
             +> genReturnType
             +> (fun ctx -> genEqualsInBinding (equalsRange ctx) ctx)
 
-        let short = prefix +> genExprKeepIndentInBranch astContext e
+        let short =
+            prefix +> genExprKeepIndentInBranch astContext e
+            |> genTriviaForReturnType
```

Comment:
This is wrong, you are printing the trivia after a different node.

I had to shuffle things around a bit, but this is more accurate:
```fsharp
    let genEqualsInBinding (ctx: Context) =
        let equalsRange =
            let endPos =
                match returnType with
                | Some rt -> rt.Range.End
                | None -> valueName.Range.End

            ctx.MkRange endPos e.Range.Start
        let space =
            ctx.TriviaTokenNodes
            |> Map.tryFindOrEmptyList EQUALS
            |> fun triviaNodes ->
                match TriviaHelpers.findInRange triviaNodes equalsRange with
                | Some tn when (List.isNotEmpty tn.ContentAfter) -> sepNone
                | _ -> sepSpace

        (tokN equalsRange EQUALS (sepSpace +> sepEqFixed) +> space) ctx
    
    let genReturnType =
        match returnType with
        | Some rt ->
            let hasGenerics =
                match valueName with
                | SynPat.LongIdent (_, _, Some _, _, _, _) -> true
                | _ -> false

            ifElse hasGenerics sepColonWithSpacesFixed sepColon
            +> (genType astContext false rt |> genTriviaFor SynBindingReturnInfo_ rt.Range)
            +> autoIndentAndNlnWhenWriteBeforeNewlineNotEmpty genEqualsInBinding
        | None -> genEqualsInBinding
```

I print the `=` as part of the `genReturnType` because it should be indented if there is a comment after the return type.

`autoIndentAndNlnWhenWriteBeforeNewlineNotEmpty` can seem a bit weird.
When we have code comment that comes after a node we don't add them directly as a `Write` event in the Context.

Reply:
Ah, thanks man.
I went with a slightly different format in the test. After reviewing other `EQUALS` that occur as the only character in a line, I think the `LBRACK` should be placed in a vertical line with the `EQUALS` without further indent.